can/interfaces/vector/canlib.py

```diff
@@ -93,6 +97,24 @@ def __init__(self, channel, can_filters=None, poll_interval=0.01,
         # Get channels masks
         self.channel_masks = {}
         self.index_to_channel = {}
+
+        if serial is not None:
+            app_name = None
+            channel_index = []
+            channel_configs, channel_count = get_channel_configs()
```

Comment:
Should probably move `vxlapi.xlOpenDriver()` above to after `get_channel_configs()` as it calls `vxlapi.xlCloseDriver()`.